core/src/main/java/io/questdb/griffin/engine/groupby/GroupByUtils.java

```diff
@@ -613,4 +611,19 @@ private static boolean compareNodesGroupByFunctionKey(
 
         return false;
     }
+
+    /**
+     * Returns the column index from the given base metadata in two cases:
+     * 1. it's a column literal
+     * 2. cast(a_column as same_type) which is compiled as a_column (the case is no-op)
+     */
+    private static int findColumnKeyIndex(ExpressionNode node, Function func, RecordMetadata baseMetadata) {
+        if (node.type == LITERAL) {
+            return baseMetadata.getColumnIndexQuiet(node.token);
+        }
+        if (SqlKeywords.isCastKeyword(node.token) && func instanceof ColumnFunction) {
```

Comment:
a wild guess: will this work with JSONs? `SqlOptimizer` does some sorcery with JSON and casting.